key-export-client/tests/interpolate_secret_key.rs

```diff
@@ -0,0 +1,95 @@
+use dfns_key_export_client::{interpolate_secret_key, InterpolateKeyError};
+use dfns_key_export_common::KeySharePlaintext;
+use generic_ec::{Curve, NonZero, Point, Scalar, SecretScalar};
+
+fn get_random_keys_and_shares<E: Curve>(
+    t: u16,
+    n: u16,
+) -> (SecretScalar<E>, Point<E>, Vec<KeySharePlaintext<E>>) {
+    let mut rng = rand_dev::DevRng::new();
+
+    let secret_key = SecretScalar::<E>::random(&mut rng);
+    let public_key = Point::generator() * &secret_key;
+
+    let secret_shares = {
+        let key_shares_indexes = (1..=n)
+            .map(|i| NonZero::from_scalar(Scalar::from(i)))
+            .collect::<Option<Vec<_>>>()
+            .unwrap();
+        let f = generic_ec_zkp::polynomial::Polynomial::sample_with_const_term(
+            &mut rng,
+            usize::from(t) - 1,
+            secret_key.clone(),
+        );
+        let secret_shares = key_shares_indexes
+            .iter()
+            .map(|i| f.value(i))
+            .map(|mut x| SecretScalar::new(&mut x))
+            .collect::<Vec<_>>();
+        secret_shares
+    };
+
+    let shares = (1..n + 1)
```

Comment:
Not critical, but you can use 
```suggestion
    let shares = (1..=n)
```